### PR TITLE
Workaround for trace infinity loop in exception handling

### PR DESCRIPTION
**Pull request by marvinhinz**
I copied the workaround from #54 because the pr is not yet splitted and i needed the functionality.

### CODE REVIEW

**Comment by tmotyl** [comment]
Ive got the counter solution, looks ok, but why did you also change the dirname part?

**Comment by marvinhinz** [comment]
> Ive got the counter solution, looks ok, but why did you also change the dirname part?

Oops, didnt see this while copying. I'll fix it

File: magmi/inc/magmi_engine.php [hunk 1]
```diff
@@ -407,6 +408,10 @@ public function getExceptionTrace($tk, &$traces)
                     $trstr .= "\n";
                 }
             }
+            // simple max trace depth fix
+            if (++$counter % 20) {
```
Comment (tmotyl) [inline magmi/inc/magmi_engine.php]: Have you tested the code?
By reading it looks like the if will break on every counter value except 20 and multiplications of 20, right? Shouldn't there be a negation here?
1 % 20 == 1 => true
2 % 20 == 2 => true
19 % 20 == 19 => true
20 % 20 == 0 => false
21 % 20 == 1 => true

Comment (marvinhinz) [reply magmi/inc/magmi_engine.php]: You're right, sorry! This should be compared against zero.

File: magmi/inc/magmi_engine.php [hunk 2]
```diff
@@ -409,7 +409,7 @@ public function getExceptionTrace($tk, &$traces)
                 }
             }
             // simple max trace depth fix
-            if (++$counter % 20) {
+            if (++$counter % 20 == 0) {
```
Comment (tmotyl) [inline magmi/inc/magmi_engine.php]: lets have === just for a sake of being clean.

Comment (marvinhinz) [reply magmi/inc/magmi_engine.php]: Done :)

**Comment by tmotyl** [comment]
thanks!

**Pull request merged**